### ISSUE TITLE
Extend article rating support to additional wikis

### ISSUE THREAD

**Issue by ragesoss** [WikiEducationFoundation/WikiEduDashboard#1851]
The PageAssessements API of Wikipedia is now enabled on Arabic and Turkish Wikipedias ([with more to come](https://phabricator.wikimedia.org/T120219)). RatingImporter should add support for those wikis.

This will probably make sense to implement after #1834 is complete.

**Comment by MusikAnimal** [comment]
Hungarian and French are now supported, too.

I don't know how you expose assessments in the Dashboard, but if you care about the wiki's preferred colours and icons for classification/importance levels, you may be interested in the XTools API for this: https://xtools.wmflabs.org/api/project/assessments I am committed to keeping this up-to-date, so you can programatically know which wikis are supported. This is a cheap endpoint but do try to cache the result if you would otherwise be making a buttload of requests very quickly :)

**Comment by ayaankazerouni** [comment]
Hi @ragesoss, is this still needed? If so, I'd like to help.

My current plan is to update the `en_wiki` query (and method name) in `rating_importer.rb` to give back all the supported wikis, and then change the various `update_rating` methods to perform updates on a per-wiki basis.

* Is there anything you think I should watch out for or be aware of?
* How would you like to keep track of which wikis are supported by the PageAssessments API? Did you want to find it programmatically as @MusikAnimal suggests, or just a listing within the importer to start with?

**Comment by ragesoss** [comment]
@ayaankazerouni yes, I think no one has tackled this yet. Feel free to request an Slack invite if you'd like to chat about it.

**Comment by nexpectArpit** [comment]
@ragesoss  can i work on this issue?

**Comment by ragesoss** [comment]
@nexpectArpit sure

**Comment by nexpectArpit** [comment]
Hi  @ragesoss 

Before beginning work on this Issue, I want to confirm , After reviewing #1851 and the related blocker #1834, the scope looks broader than a simple wiki toggle.

Current understanding:
- The article rating system is hardcoded for English Wikipedia. Ratings like Stub, Start, GA, FA rely on backend logic tied to enwiki only. Supporting ar, tr, and others requires backend changes, not frontend config alone.

- Performance blocker from #1834: RatingImporter currently takes around 4 hours for English Wikipedia. Extending to multiple wikis without changes would slow or break the Daily Update.
I traced a working solution in AverageViewsImporter, which uses Concurrent::FixedThreadPool to fetch data in parallel. Applying the same pattern to RatingImporter seems required before scaling.

- Instead of hardcoding supported wikis, I plan to use the XTools PageAssessments API to dynamically discover which wikis support ratings. This allows automatic support for future wikis once enabled upstream.

Does i am going in right direction for solving #1851 while respecting the constraints raised in #1834?

**Comment by ragesoss** [comment]
@nexpectArpit I'm not entirely sure what the best approach is here.

I think the right place to start would be to add support for on-demand fetching of ratings for articles in supported languages. We do this for English already at the time an article is assigned, via AssigmenntManager. I suggest a first pass at this just involves adding support for the one-article-at-a-time ratings import via AssignmentManager. We don't need to incorporate other languages into the `.update_all_ratings` method that gets used in the DailyUpdate.

For dynamically discovering supported languages, that seems reasonable, but it should probably involve aggressively caching those results so that we only re-check for new languages every once in a while (eg, once per week).
